packages/abstract-core/src/objects/oracle.rs

```diff
@@ -60,10 +62,23 @@ impl<'a> Oracle<'a> {
                 "Oracle list size limit exceeded",
             )));
         }
-        // remove assets from oracle
-        self.remove_assets(deps.branch(), ans, to_remove)?;
+
+        let mut all: Vec<AssetEntry> = to_add
+            .iter()
+            .map(|(a, _)| a.clone())
+            .collect::<Vec<AssetEntry>>();
+        all.extend(to_remove.clone());
+        all.dedup();
+        if all.len() != to_add.len() + to_remove.len() {
+            return Err(crate::AbstractError::Std(StdError::generic_err(
+                "Duplicate assets in update",
+            )));
+        }
+
```

Comment:
We should be able to make this more efficient and read better by leveraging a `HashSet` and iterating which should be more efficient than dedup (dedup sorts first, hence O(n log n).

```rust
    let to_add_entries: HashSet<_> = to_add.iter().map(|(a, _)| a).collect();

 for item in to_remove {
        if to_add_entries.contains(item) {
            return Err(StdError::generic_err("Duplicate assets in update"));
        }
    }
```

Reply:
haha yeah, i tried, AssetInfo doesn't implement Hash.

Reply:
also hashing is expensive and the set's we're working with are small